datalad/ui/__init__.py

```diff
@@ -81,7 +83,9 @@ def set_backend(self, backend):
                 else:
                     backend = 'dialog'
             else:
-                backend = 'dialog' if is_interactive() else 'no-progress'
+                backend = 'dialog' if is_interactive else 'no-progress'
+        if not is_interactive and backend == 'annex':
+            backend = 'annex-no-dialog'
```

Comment:
> unfortunately this would break entirely ability of users to authenticate when possible

It doesn't break the ability, it changes it - you'd need to set the new config (as env var ATM). Because we are unable to correctly detect. And as far as I can tell, it's simply not possible. (see #7345)
Keeping the behavior of assuming interactivity by default, is not an option IMO, though. Because in non-interactive jobs, that just means stalling with no error reporting whatsoever. Users would have no clue at all.
Whereas in your case, there should at least be a hint that authentication wasn't possible. I'd rather err on the somewhat informative side than on the one that leaves users with nothing to act upon.

We could maintain the behavior to some extend, when we address #7352. Passing down the (accidently correct) detection from the super process, would lead to the same behavior in your case. I'd still argue it's wrong, since detection outside of a special remote process is not more correct. It's just slightly rarer to be an issue. (see #7354)

Reply:
> It doesn't break the ability enitrely, it changes it - you'd need to set the new config (as env var ATM).

well, *not entirely* means that it still breaks it regardless how you name it.

Not breaking *at all* would be whenever the default behavior stays as is at least for `maint`. Eg. could be done via config variable staying "auto" by default and then switch to `True` or `False` based on the `utils.is_interactive()`, while keeping using `is_interactive()` function, and allowing all CIs or whatnot which knows that there is no agent behind keyboard to say so.  That would IMHO be better since it would be explicit and useful regardless on how we decide to trick/handle interactivity.    Then in `master` we could be deliberated further or even change behavior.

> Whereas in your case, there should at least be a hint that authentication wasn't possible.

why *wasn't possible* if it was and was working just fine before this change?

Reply:
> Eg. could be done via config variable staying "auto" by default and then switch to True or False based on the utils.is_interactive(), while keeping using is_interactive() function

That is what is happening. The special remote process evaluates the detection in the state of this PR. But detection (`auto`) simply doesn't work. It only happens to do the right thing, when the process in question is the top-level process. But we are querying from within the special remote process.

> why wasn't possible if it was and was working just fine before this change?

I didn't "work" before that change. It happened to do the right thing in interactive sessions and the wrong thing in non-interactive ones, b/c it always assumed to be in an interactive one.